clientv2/client.go

```diff
@@ -64,14 +65,23 @@ type Request struct {
 }
 
 // NewClient creates a new http client wrapper
-func NewClient(client *http.Client, baseURL string, interceptors ...RequestInterceptor) *Client {
-	return &Client{
+func NewClient(client *http.Client, baseURL string, options *Options, interceptors ...RequestInterceptor) *Client {
+	c := &Client{
 		Client:  client,
 		BaseURL: baseURL,
 		RequestInterceptor: ChainInterceptor(append([]RequestInterceptor{func(ctx context.Context, requestSet *http.Request, gqlInfo *GQLRequestInfo, res interface{}, next RequestInterceptorFunc) error {
 			return next(ctx, requestSet, gqlInfo, res)
 		}}, interceptors...)...),
 	}
+
+	if options != nil {
+		c.ParseDataWhenErrors = options.ParseDataWhenErrors
+	}
+	return c
+}
+
+type Options struct {
+	ParseDataWhenErrors bool
```

Comment:
Can't we achieve this with a dedicated interceptor?

Reply:
not as far as i see. the request and unmarshalling happens in `client.do`. And when error is happened data is never unmarshalled so i couldn't retreive it through the interface

Reply:
I would like to make this a little more generic in form. Because I can envision a lot of different patterns here



https://github.com/Yamashou/gqlgenc/issues/162

I thought it would be better to receive it in interface and insert the value.

Reply:
didn't really get what you mean.
`Options` argument in `NewClinet` should be an interface instead of struct?
But that means that the consumer will have to implement that interface and put the implementation. + he'll have not needed stuff in his code
Or maybe i got it wrong?
Can you at least guide me the path you mean a bit?

Reply:
My ideas

1. Allow Response type to be specified
2. Additional required types are obtained through interceptor by preparing another one.

If you don't do these things, you'll end up with more fields for various API reasons.

Reply:
@Metalscreame 
I may have been mistaken.
Does this mean that you want to return the result as it is in the event of an error?

Reply:
yes, I want to return a parsed result when error occur as in some cases both `data` and `errors` are presented in the same response from the GQL server. 
Even one of your's test cases had this example in the constant, it was `gqlDataAndErr`. I literally have the same now and need both. (it's covered with new test)

```
{
	"data": {
		"something": "some data"
	},
	"errors": [{
		"path": ["query GetUser", "viewer", "repositories", "nsodes"],
		"extensions": {
			"code": "undefinedField",
			"typeName": "RepositoryConnection",
			"fieldName": "nsodes"
		},
		"locations": [{
			"line": 6,
			"column": 4
		}],
		"message": "Field 'nsodes' doesn't exist on type 'RepositoryConnection'"
	}]
}
```

I'm working with Braintree GQL API and it returns data along with errors. 

from their doc:
```
It is possible to have partially successful responses, where **both a partially populated data object and errors are returned**. If errors prevent a field in your query from resolving, the field in the data object will be returned with the value null and relevant errors will be in the error object.
```